runtime/compiler/x/env/J9CPU.cpp

```diff
@@ -24,71 +24,67 @@
 #include "env/CompilerEnv.hpp"
 #include "env/CPU.hpp"
 #include "env/VMJ9.h"
+#include "x/runtime/X86Runtime.hpp"
+#include "env/JitConfig.hpp"
+
+// This is a workaround to avoid J9_PROJECT_SPECIFIC macros in x/env/OMRCPU.cpp
+// Without this definition, we get an undefined symbol of JITConfig::instance() at runtime
+TR::JitConfig * TR::JitConfig::instance() { return NULL; }
 
 namespace J9
 {
 
 namespace X86
 {
 
-const char *
-CPU::getX86ProcessorVendorId(TR::Compilation *comp)
+TR_X86CPUIDBuffer *
+CPU::queryX86TargetCPUID()
    {
-   return comp->fej9()->getX86ProcessorVendorId();
+   static TR_X86CPUIDBuffer buf = { {'U','n','k','n','o','w','n','B','r','a','n','d'} };
+   jitGetCPUID(&buf);
+   return &buf;
    }
 
-uint32_t
-CPU::getX86ProcessorSignature(TR::Compilation *comp)
-   {
-   return comp->fej9()->getX86ProcessorSignature();
-   }
-
-uint32_t
-CPU::getX86ProcessorFeatureFlags(TR::Compilation *comp)
-   {
-   return comp->fej9()->getX86ProcessorFeatureFlags();
-   }
-
-uint32_t
-CPU::getX86ProcessorFeatureFlags2(TR::Compilation *comp)
+const char *
+CPU::getProcessorVendorId()
    {
-   return comp->fej9()->getX86ProcessorFeatureFlags2();
+   return self()->getX86ProcessorVendorId();
    }
 
 uint32_t
-CPU::getX86ProcessorFeatureFlags8(TR::Compilation *comp)
+CPU::getProcessorSignature()
    {
-   return comp->fej9()->getX86ProcessorFeatureFlags8();
+   return self()->getX86ProcessorSignature();
    }
 
 bool
-CPU::testOSForSSESupport(TR::Compilation *comp)
-   {
-   return comp->fej9()->testOSForSSESupport();
-   }
-
-void
-CPU::setX86ProcessorFeatureFlags(uint32_t flags)
+CPU::testOSForSSESupport()
```

Comment:
This can be removed if https://github.com/eclipse/omr/pull/2822 is merged before this PR gets merged.